lms/djangoapps/experiments/flags.py

```diff
@@ -170,12 +185,19 @@ def is_enabled_without_course_context(self):
         return self.is_enabled()
 
     def is_experiment_on(self, course_key=None):
-        return super().is_enabled(course_key)
+        # If no course_key is supplied check the global flag irrespective of courses
+        if course_key is None:
+            return super().is_enabled_without_course_context()
```

Comment:
Instead of doing this "if else" stuff, you could just replace this method with:
```
return super()._is_enabled(course_key)
```

Which will correctly handle None. I know it's a private method, but we already use it elsewhere in this class. So what's one more sin?  (Actually, that other place we call `_is_enabled`, could be replaced with a call to this method here.)

But also, your "if else" way is fine too and probably more correct. I have an over-bias towards brevity.

Reply:
Thanks, I left it verbose, but swapped the other call to `_is_enabled` to `is_experiment_on`